pkg/contract/verifier.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Nuts auth
+ * Copyright (C) 2020. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package contract
+
+// State contains the outcome of the verification. It van be VALID or INVALID. This makes it human readable.
+type State string
+
+const (
+	// Valid is used to indicate a contract was valid on the time of testing
+	Valid State = "VALID"
+	// Invalid is used to indicate a contract was invalid on the time of testing
+	Invalid State = "INVALID"
+)
+
+// VerifierType is the type for a specific verifier
+type VerifierType string
+
+// Verifier defines the funcs needed to verify a VerifiablePresentation
+type Verifier interface {
+	// ValidateVP validates a verifiable presentation, it's up to the caller to select the right verifier for the given VerifiablePresentation type
+	VerifyVP(rawVerifiablePresentation []byte) (*VerificationResult, error)
+}
+
+// VerifiableCredentialBase holds the basic fields for a VerifiableCredential
+// todo: move or use lib
+type VerifiableCredentialBase struct {
+	Context      []string `json:"@context"`
+	ID           *string
+	Type         []string
+	Issuer       *string
+	IssuanceDate *string
```

Comment:
If this is a date/time, I'd declare it as `time.Time` and provide a unmarshaler if necessary